### PR TITLE
Additional modification in MIRCO to use as a Library

### PR DESCRIPTION
**Pull request by RShaw026**
## Description and Context

In this MR, variables are renamed to more meaningful names. Some unnecessary variables and parts of codes are removed. 

`mirco_` prefix is added to all the header and CPP files so that the headers are easily distinguishable while using MIRCO as a library. 

`std::shared_ptr` is changed to `Teuchos::RCP` for the `surfacegenerator` object.

`meshgrid` is passed as a reference for optimization.

## Related Issues and Pull Requests

* Closes
* Blocks
* Is blocked by
* Follows MR #64 
* Precedes
* Related to
* Part of
* Composed of

## How Has This Been Tested?

`ctest` is working. Also, the generated build files are imported into BACI, and a simple block-over-block problem is solved in BACI using MIRCO functions.

## Checklist

- [x] My commit messages mention the appropriate GitHub issue numbers and are longer than one line where appropriate.
- [x] I have added/updated documentation where necessary.

## Interested Parties / Possible Reviewers

@mayrmt @NoraHagmeyer

### CODE REVIEW

File: src/main.cpp
Comment (mayrmt) [inline src/main.cpp:32]: It's good practice to initialize variables right at declaration. If not, you rely on `MIRCO::SetParameters()` to properly initialize all variables. Of course, setting the desired parameter from input should happen in `MIRCO::SetParameters()`, but initialization could happen before (so that you have maybe wrong values, but at least _a_ value for each variable.

File: src/mirco_evaluate.cpp
Comment (mayrmt) [inline src/mirco_evaluate.cpp:14]: I should not use this without limiting its scope. In this particular case, it does not matter that much, because it is in a cpp-file. However, if such a statement shows up in a header file, then it propagates through header including. 

I recommend to never use this outside tiny scopes to make sure that it never propagates. In fact, I'd never write `using namespace std`. In most cases, it's sufficient to list the two or three objects from the STL that you really want to use, e.g.
```c++
using std:cout
using std::vector
```
but only inside each function, where you want to use them (not a global scope).

File: src/mirco_filesystem_utils.cpp
Comment (mayrmt) [inline src/mirco_filesystem_utils.cpp:4]: Maybe, this could be achieved or even replaced with stuff from `std::filesystem` in the future.

**Review by mayrmt — DISMISSED** [review]
Thanks! Looks good.

**Comment by RShaw026** [comment]
@mayrmt I need to make one more change before merging this branch. `meshgrid` needs to be passed reference in the `Evaluate` function. I want to push it here after ensuring it works with BACI without any issues.

**Comment by mayrmt** [comment]
@RShaw026 Sure! 

I approve if it's ok for me. That does not mean that you have to merge. If you want to do further changes, feel free to do so.

**Review by mayrmt — APPROVED** [review]

**Pull request merged**
